### DISCUSSION TITLE
Walking Pattern Generator Example

### DISCUSSION THREAD

**Discussion by NoePity2** [stephane-caron/pink#105] [General]
Hello, Stephane. What does Pink need for an example of a walking pattern generator like the one in Pymanoid?

Greetings

**Comment by stephane-caron** [comment]
A nice contributor spending the time to adapt [the previous example](https://github.com/stephane-caron/pymanoid/blob/d4e48c302e68127f4878e02fdc209f607328ee61/examples/horizontal_walking.py) :slightly_smiling_face: 

The main parts are out there already:

- The differential IK with foot-frame and CoM tasks in Pink, see for instance [`humanoid_g1_com.py`](https://github.com/stephane-caron/pink/blob/5491beb66cb09d43805be74f8ec191e198fb106c/examples/humanoid_g1_com.py)
- The linear MPC in the [`lipm_walking_controller.py`](https://github.com/stephane-caron/qpmpc/blob/main/examples/lipm_walking_controller.py) example of [qpmpc](https://github.com/stephane-caron/qpmpc/) to generate the CoM trajectory

I'd be happy to help review and merge a PR for this.

**Reply by NoePity2** [reply]
Thank you for your reply!

We have a visual servoing implementation of my PhD that gives us the CoM Trajectory and the footsteps in  [libvwpg](https://github.com/cimat-ris/libvwpg). This library is written in C++, but we have Python bindings. Previously, we used _Pymanoid_ with this library and published a  [paper](https://www.sciencedirect.com/science/article/abs/pii/S0921889019306694). 

I would happily merge our visual walking pattern generator with the differential IK with foot-frame and CoM tasks in Pink to create the walking example and continue researching humanoid robots.
